task_geo/common/country_codes.py

```diff
@@ -469,9 +468,152 @@
+                    'JM': 'JAM', 'JP': 'JPN', 'JE': 'JEY', 'JO': 'JOR', 'KZ': 'KAZ', 'KE': 'KEN',
+                    'KI': 'KIR', 'KR': 'KOR', 'KW': 'KWT', 'KG': 'KGZ', 'LA': 'LAO', 'LV': 'LVA',
+                    'LB': 'LBN', 'LS': 'LSO', 'LR': 'LBR', 'LY': 'LBY', 'LI': 'LIE', 'LT': 'LTU',
+                    'LU': 'LUX', 'MO': 'MAC', 'MK': 'MKD', 'MG': 'MDG', 'MW': 'MWI', 'MY': 'MYS',
+                    'MV': 'MDV', 'ML': 'MLI', 'MT': 'MLT', 'MH': 'MHL', 'MQ': 'MTQ', 'MR': 'MRT',
+                    'MU': 'MUS', 'YT': 'MYT', 'MX': 'MEX', 'FM': 'FSM', 'MD': 'MDA', 'MC': 'MCO',
+                    'MN': 'MNG', 'ME': 'MNE', 'MS': 'MSR', 'MA': 'MAR', 'MZ': 'MOZ', 'MM': 'MMR',
+                    'NA': 'NAM', 'NR': 'NRU', 'NP': 'NPL', 'NL': 'NLD', 'AN': 'ANT', 'NC': 'NCL',
+                    'NZ': 'NZL', 'NI': 'NIC', 'NE': 'NER', 'NG': 'NGA', 'NU': 'NIU', 'NF': 'NFK',
+                    'MP': 'MNP', 'NO': 'NOR', 'OM': 'OMN', 'PK': 'PAK', 'PW': 'PLW', 'PS': 'PSE',
+                    'PA': 'PAN', 'PG': 'PNG', 'PY': 'PRY', 'PE': 'PER', 'PH': 'PHL', 'PN': 'PCN',
+                    'PL': 'POL', 'PT': 'PRT', 'PR': 'PRI', 'QA': 'QAT', 'RE': 'REU', 'RO': 'ROU',
+                    'RU': 'RUS', 'RW': 'RWA', 'BL': 'BLM', 'SH': 'SHN', 'KN': 'KNA', 'LC': 'LCA',
+                    'MF': 'MAF', 'PM': 'SPM', 'VC': 'VCT', 'WS': 'WSM', 'SM': 'SMR', 'ST': 'STP',
+                    'SA': 'SAU', 'SN': 'SEN', 'RS': 'SRB', 'SC': 'SYC', 'SL': 'SLE', 'SG': 'SGP',
+                    'SK': 'SVK', 'SI': 'SVN', 'SB': 'SLB', 'SO': 'SOM', 'ZA': 'ZAF', 'GS': 'SGS',
+                    'ES': 'ESP', 'LK': 'LKA', 'SD': 'SDN', 'SR': 'SUR', 'SJ': 'SJM', 'SZ': 'SWZ',
+                    'SE': 'SWE', 'CH': 'CHE', 'SY': 'SYR', 'TW': 'TWN', 'TJ': 'TJK', 'TZ': 'TZA',
+                    'TH': 'THA', 'TL': 'TLS', 'TG': 'TGO', 'TK': 'TKL', 'TO': 'TON', 'TT': 'TTO',
+                    'TN': 'TUN', 'TR': 'TUR', 'TM': 'TKM', 'TC': 'TCA', 'TV': 'TUV', 'UG': 'UGA',
+                    'UA': 'UKR', 'AE': 'ARE', 'GB': 'GBR', 'US': 'USA', 'UM': 'UMI', 'UY': 'URY',
+                    'UZ': 'UZB', 'VU': 'VUT', 'VE': 'VEN', 'VN': 'VNM', 'VG': 'VGB', 'VI': 'VIR',
+                    'WF': 'WLF', 'EH': 'ESH', 'YE': 'YEM', 'ZM': 'ZMB', 'ZW': 'ZWE', 'KP': 'PRK',
+                    'SS': 'SSD'}
+
+
+def iso3_to_iso2(country_iso3):
+    """Converts iso3 to iso2 country codes
+
+    See Also: https://www.iban.com/country-codes
+
+    Args:
+        country_iso3 : str
+
+    Returns:
+        str
+
+    """
+    return ISO3_TO_ISO2_MAP[country_iso3]
+
+
+def iso2_to_iso3(country_iso2):
+    """Converts iso2 to iso3 country codes
+
+    See Also: https://www.iban.com/country-codes
+
+    Args:
+        country_iso2 : str
+
+    Returns:
+        str
+
+    """
+    return ISO2_TO_ISO3_MAP[country_iso2]
+
 
-def iso_to_name(isocode):
-    return ISO_CODES_COUNTRY_NAME_MAP.get(isocode)
+def iso2_to_name(country_iso2):
+    return ISO_CODES_COUNTRY_NAME_MAP.get(country_iso2)
```

Comment:
This is, as code, correct as it could be, but as a concept is wrong, let me explain why:

I did it first for the FIPS codes and knew that something was not good, but I didn't realize **what**  after just seeing your submission, so the only one to blame here is me, you only followed my example. I must, however, ask you to help me fix my past mistakes.

The main error is that we should not have a dictionary for each code mapping, but a single reference, containing all the codes, be them FIPS, ISO2, ISO3, or whatever we could find in the future.

How we do this?

1. First of all we need to merge all dictionaries in this file in a `pandas.DataFrame`  with columns `['name', 'fips', 'iso2', 'iso3']` containing all the values of the dics we have collected so far.
We must accept that some countries will have different names in the FIPS and ISO references, so we need to check manually; and also, that some that are present in one reference will not be in the other, and leave those values blank.

After we have that dataframe, we should save it with
 ```python
df.to_csv('task_geo/common/country_codes.csv', index=False, header=True)
```

2. Now that we have this reference we should import it at top-module level, that is:

```python
# In task_geo/common/country_codes.py
import pandas as pd

COUNTRY_CODES = pandas.read_csv('country_codes.csv')
```

3. Now that we have our references in the module, we will write a function that allow us to query it to translate codes, it should be something like this:

```python

def translate_code(value, input, output):
    """Translate code `value` from `input` format to `output`.

    Arguments:
        value(str): Value to translate.
        input(str): Format the value is in, possible values are: ['name', 'iso2', 'iso3','fips']
       output(str): Format the value should be returned, possible values are:['name', 'iso2', 'iso3','fips']

    Returns:
        str

    """
    # This lookup should work, but the .values[0] is not very elegant, if you find a way to improve it, please go ahead.
    return COUNTRY_CODES[COUNTRY_CODES[input] == value][output].values[0]
```

4. Now we can rewrite our functions (iso2_to_name, ...) using one-liner calls to our `translate_code`, we append an x at the beginning of the name, the functions duplicated.

5. We check that the outputs returned by our functions are the same ( minus the name corrections that we could have done) by checking the the values for each key in the old references. This will help us spot if something went wrong while organizing the reference file.

6. We delete the older functions and references.

Reply:
@ManuelAlvarezC Please have a look at the country code translator and let me know if anything needs to be changed. This is a draft as of now.